app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java

```diff
@@ -164,11 +173,107 @@ public static GoogleAuthInfo parseUri(Uri uri) throws GoogleAuthInfoException {
         return new GoogleAuthInfo(info, accountName, issuer);
     }
 
+    public static Export parseExportUri(String s) throws GoogleAuthInfoException {
+        Uri uri = Uri.parse(s);
+        if (uri == null) {
+            throw new GoogleAuthInfoException("Bad URI format");
+        }
+        return GoogleAuthInfo.parseExportUri(uri);
+    }
+
+    public static Export parseExportUri(Uri uri) throws GoogleAuthInfoException {
+        String scheme = uri.getScheme();
+        if (scheme == null || !scheme.equals(SCHEME_EXPORT)) {
+            throw new GoogleAuthInfoException("Unsupported protocol");
+        }
+
+        String host = uri.getHost();
+        if (host == null || !host.equals("offline")) {
+            throw new GoogleAuthInfoException("Unsupported host");
+        }
+
+        String data = uri.getQueryParameter("data");
+        if (data == null) {
+            throw new GoogleAuthInfoException("Parameter 'data' is not set");
+        }
+
+        GoogleAuthProtos.MigrationPayload payload;
+        try {
+            byte[] bytes = Base64.decode(data);
+            payload = GoogleAuthProtos.MigrationPayload.parseFrom(bytes);
+        } catch (EncodingException | InvalidProtocolBufferException e) {
+            throw new GoogleAuthInfoException(e);
+        }
+
+        List<GoogleAuthInfo> infos = new ArrayList<>();
+        for (GoogleAuthProtos.MigrationPayload.OtpParameters params : payload.getOtpParametersList()) {
+            OtpInfo otp;
+            try {
+                byte[] secret = params.getSecret().toByteArray();
+                switch (params.getType()) {
+                    case OTP_HOTP:
+                        otp = new HotpInfo(secret, params.getCounter());
+                        break;
+                    case OTP_TOTP:
+                        otp = new TotpInfo(secret);
+                        break;
+                    default:
+                        throw new GoogleAuthInfoException(String.format("Unsupported algorithm: %d", params.getType().ordinal()));
+                }
+            } catch (OtpInfoException e){
+                throw new GoogleAuthInfoException(e);
+            }
+
+            String name = params.getName();
+            String issuer = params.getIssuer();
+            int colonI = name.indexOf(':');
+            if (issuer.isEmpty() && colonI != -1) {
+                issuer = name.substring(0, colonI);
+                name = name.substring(colonI + 1);
+            }
+
+            GoogleAuthInfo info = new GoogleAuthInfo(otp, name, issuer);
```

Comment:
I think the account descriptions from Google Authenticator will suit better in our issuer field instead of the account name field since users will more likely enter the Issuer in Google Authenticator.

Reply:
I'm not sure that's a good idea. I'd propose only doing that if the given issuer is empty. Otherwise we'd be doing the wrong thing in most cases.

Reply:
My bad, you're right. This is fine!